### ISSUE TITLE
Undo, and a few other issues

### ISSUE THREAD

**Issue by KjellConnelly** [vreelb/dewdle#5]
Hey I noticed you use fabricjs, and I think they have an undo function. Is there a way to add this as a keystroke? Would be cool to undo instead of clearing the entire canvas. 

Also a few other things I noticed that maybe should be mentioned:

1) 
If you try to open the JSON file using the  file:// protocol on Chrome, it fails, so you need to add `--allow-file-access-from-files`. I created a bash script for my mac:

```bash
open /Applications/Google\ Chrome.app --args --allow-file-access-from-files
open file:///Users/macbook-pro/Desktop/dewdle-master/draw.html
open file:///Users/macbook-pro/Desktop/dewdle-master/control.html
open file:///Users/macbook-pro/Desktop/dewdle-master/render.html
```

2) If you open `index.html`, the links fail because they try to open `/render` instead of `/render.html`. Same with draw/control.html

3) My render.html page wouldn't work. Had to open `css/render.css` and remove `display:none`.

4) In order to get a simple keying, I had to goto `css/ui.css` and remove everything from `#canvas-contain`. This css created a grid. In order to deal with this using OBS, I didn't see a simply way to key out a grid (maybe you know of one I don't know of?) So it was easier to just key out the color `white` when there is no background color. I suppose I could've keyed out 2 colors, but why use more processing power to key out 2 colors using 2 filters when it could be done with one color and one filter?

Thanks for this project btw :) Really cool stuff!

**Comment by vreelb** [comment]
Thanks for checking out this project! I haven't done much with it in a while so I'm glad to see there's still interest.

Regarding 1-3, I haven't looked into loading the files directly with `file:///` since early on and it hasn't really a priority. In the meantime accessing those pages through localhost on your configured port should work as intended (I just tested it now, see image below). The simplest solution is probably for me to better document how to get started.

Regarding 4 no keying should be necessary. The embedded browser's background is transparent by default (though you can change the CSS of this in OBS I believe).
![image](https://user-images.githubusercontent.com/1118472/80839103-8f2c8100-8bc8-11ea-8245-9d051d8c2cb1.png)

Undo is a feature I've had in mind, but unfortunately since we're not just dealing with a single page (possibly N different draw and control pages) it gets more complicated. The logic for connection-loss and recovery complicates this further as there are a lot of canvas states that can get broadcast around without user input. Given that the connection-loss stuff is probably more important for live TV rather than running it locally while streaming I should probably make it configurable, which would make the undo logic simpler.

Thanks again for reaching out, I may putter around with this a bit more now that I know people are still stumbling across it.

**Comment by KjellConnelly** [comment]
Hey vreelb, thanks for the quick reply :) I did a bit of investigating and realized that I was opening the html files using Finder instead of `localhost:12180`, hence why I had trouble opening using the `file://` protocol. Also yes, you're right about #4, no keying is necessary. I guess either the `display:none` was a problem from before, so I started getting the browser input to OBS from draw.html at some point since render.html wasn't working. Then I forgot to uncomment the transparency grid. 

On OBS, I was also getting my browser url from a local file (/Users/mini/..../render.html) instead of `http://localhost:12180/render.html`.

Bunch of little mistakes. Anyways, you might want to add this to your npm scripts, something like: 
```JSON
"scripts": {
    ...,
    "start": "node server.js & open http://localhost:12180"
  },
```

Then run `npm start` instead of `node server.js` since it will start the server and the browser in one go now.
And also add something to the README file where it tells to link to the localhost url, and not using the local filesystem. 

Finally, onto the undo issue. Thanks for explaining it :) I understand it's more complicated than just adding a responder function now. If you get around to it, that'd be cool. If not, no worries :) I got it running on my iPad via sidecar and using an Apple Pencil to do overlays now which is cool.
